### DISCUSSION TITLE
Life hook livewire 2.11

### DISCUSSION THREAD

**Discussion by nerocil** [livewire/livewire#5538] [1. Help] [answered]
Try to leaning Livewire  but when it try trigger  updated life hook not work 
example 
`public function updatedName($name){
       dd($this->name);
}`

and I find this message from livewire doc.
> Please note that mutating a property directly inside a Livewire component class doesn't trigger any of the updating/updated hooks.

Please. do someone can explain in detail this message from [https://laravel-livewire.com/docs/2.x/lifecycle-hooks](https://laravel-livewire.com/docs/2.x/lifecycle-hooks) meaning and what should be done instead

**Comment by joshhanley** [comment]
@nerocil how are you trying to trigger the hook? Can you show your code?

**Comment by nerocil** [comment]
Okay
I try to do some thing like this 
blade
```
<div>
    <button wire:click.prevent="onDecrement" >--</button>
    <input wire:model.debounce.500ms="quantity" type="number" >
    <button wire:click.prevent="onIncrement">++</button>
</div>
```

Livewire Component

```
class ExampleComponent extends Component
{
    public int $quantity = 1;

    public function render()
    {
        return view('livewire.example-component');
    }

    function onIncrement(){
        $this->quantity ++;
    }

    function onDecrement(){
        if ($this->quantity == 1){
            return;
        }
        $this->quantity --;
    }

    //This function not triggered when buttons pressed but only when input changes

    public function updatedQuantity($quantity){
        dd($this->quantity);
    }
}
```

**Comment by joshhanley** [comment, marked as answer]
@nerocil ah right, so the updating/updated hooks only get called if the property is called from the front-end.

If you want to run one of them when updating the property on the back-end, you will need to call the method manually. This is because there is no way to detect that the property has been changed on the back-end to be able to run the method.

Hope this helps!

**Reply by nerocil** [reply]
Okay thanks

**Reply by joshhanley** [reply]
No worries! 🙂
